core_v2/sources/readme.md

```diff
@@ -0,0 +1,100 @@
+# Aptos Naming Service
+
+The Aptos Naming Service is a decentralized naming service for the Aptos blockchain.
+At its core, the naming service is an on-chain contract that allows users to register domain and subdomain names, and
+point them to an account. This allows users to have a human-readable name for their account, and to use it in place of
+the long and un-ergonomic hex addresses that exist today.
+
+For example, instead of sending a transaction to `0x1234567890abcdef1234567890abcdef1234567890abcdef1234567890abcdef`,
+you would send it to `max.apt` instead.
+
+### Terms
+
+- **Domain Name**: A domain name is a string of UTF8 characters.
+- **Subdomain Name**: A domain name is a string of UTF8 characters.
+- **Name**: When we refer to a `name` we mean either `subdomain_name` or `domain_name`.
+  This is used when describing something that applies to both `Domains` _**and**_ `Subdomains`.
+- **Character Set**: In order to allow for a safe rollout of names, we are breaking UTF8
+- **Glyph**: A single character in a character set. For example, `a` is a single glyph in the `latin` character set,
+  and `你好` is two glyphs in the `simplified_chinese` character set. We specifically use `glyph` instead of `letter` to
+  help avoid confusion when dealing with character sets where the term `letter` is not applicable, such
+  as `simplified_chinese`, or `emojis`, such as this unicorn: 🦄. Aptos is a unicorn-friendly zone. Neigh!
+
+### Character Sets
+
+With great UTF8 flexibility comes great UTF8 pain. In order to avoid this pain out of the gate, we are launching with a
+limited character set, which parallels ASCII English letters and numbers with the characters it supports. We are
+referring to
+this character set as `latin`.
+We will expand both the number of character sets we support, and the glyphs within each one, as quickly as possible to
+do safely.
```

Comment:
could we make a single line for a paragraph? it makes it easier to read in most text editors, github, and git does a good job of diffs on paragraphs now :raised_hands: